### PR TITLE
Pass SSH_AUTH_SOCK variable for ssh authentication while cloning repo…

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
…s - SR-561

https://bugs.swift.org/browse/SR-561

### CODE REVIEW

File: Sources/dep/Git.swift [hunk 1]
```diff
@@ -91,10 +91,14 @@ class Git {
         }
 
         do {
+            var environment = [String:String]()
+            if let sshAuthSockPath = getenv("SSH_AUTH_SOCK") { //for auto ssh authentication if possible
+                environment["SSH_AUTH_SOCK"] = sshAuthSockPath
```
Comment (kylef) [inline Sources/dep/Git.swift]: There are also other env variables that might be nice to replicate.

- `GIT_ASKPASS` ([git-credentials](https://git-scm.com/docs/gitcredentials))
- `SSH_ASKPASS`
- `XDG_CONFIG_HOME` (used for `git-credenitials` lookup)

In which, I wonder why not just inherit all environ variables?

Comment (aciidb0mb3r) [inline Sources/dep/Git.swift]: there is a conversation about inheriting all variables here : https://bugs.swift.org/browse/SR-588

**Comment by ddunbar** [comment]
I think we intentionally want to limit the variables in some places (during builds), but when we are talking to a third party tool like git, I agree we should probably default to inheriting the full process environment.

**Comment by aciidb0mb3r** [comment]
ok, I can convert this to inherit the full list.
What would be the best way to get the `environ` without Foundation's `NSProcessInfo` ?

**Comment by ddunbar** [comment]
We are only going to be able to do this via C, unfortunately. What we could do is try and get environ support added to the Swift stdlib (in the Process enum, next to the arguments fields). If someone wanted to push that through, its a good addition for everyone and then we can just use it.

**Comment by gribozavr** [comment]
You'd also want to inherit various environment variables that set the locale.  Git is localized to many languages.

**Comment by mxcl** [comment]
@gribozavr can you point us at a list?

**Comment by gribozavr** [comment]
All `LC_*` variables, `LANG`, and also `LANGUAGE` on Linux.

Passing through `EDITOR` and `PAGER` also makes sense, in the case `git` wants to do something interactively.

And we also want to pass through `TERM`, too.

**Comment by mxcl** [comment]
Thanks, will apply.

**Comment by aciidb0mb3r** [comment]
what is preferable? copying entire environ or only useful subset ?

**Comment by mxcl** [comment]
It is preferable to copy as little as possible as the environment can have unforeseen consequences on the execution of tools.

**Comment by kostiakoval** [comment]
@swift-ci Please test

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by kostiakoval** [comment]
is @swift-ci is only available for main swift repo?

**Comment by aciidb0mb3r** [comment]
In the meanwhile, updated to pass list of envs commented here

**Comment by mxcl** [comment]
Post my refactor this needs to be merged, I am happy to do the merge, since I caused the problem, let me know.

**Comment by aciidb0mb3r** [comment]
@mxcl Done

**Pull request merged**